application/client.plugins/projects/serial/src/lib/views/sidebar.vertical/component.ts

```diff
@@ -313,11 +315,168 @@ export class SidebarVerticalComponent implements AfterViewInit, OnDestroy {
+            if (ports) {
+                for (const port of ports) {
+                    this._addDropdownElement(port);
+                }
+            }
+        }
+    }
+
+    public _ng_changeDropdownSelect(value: string) {
+        this._chosenPort = value;
+        this._ng_ports.forEach( port => {
+            if (port.comName === value && Service.savedSession.has(this.session)) {
+                Service.savedSession.get(this.session).default = port.comName;
+            }
+        });
+    }
+
+    private _loadSession() {
+        if (Service.savedSession.has(this.session)) {
+            this._ng_defaultPort = Service.savedSession.get(this.session).default;
+            this._forceUpdate();
+        }
+    }
+
+    private _createOptions() {
+        const connectedPorts: string[] = this._ng_connected.map(connected => connected.port.comName);
+        this._ng_ports.forEach(port => {
+            if (connectedPorts.indexOf(port.comName) === -1) {
+                this._portOptions.push({path: port.comName, options: this._options.options, reader: this._options.reader});
+            }
+        });
+    }
+
+    private _removeOptions() {
+        this._portOptions = [];
+    }
+
+    private _startSpy() {
+        this._createOptions();
+        Service.startSpy(this._portOptions).catch((error: Error) => {
+            this._logger.error(error);
+        });
+    }
+
+    private _stopSpy() {
+        return new Promise((resolve) => {
+            Service.stopSpy(this._portOptions).then(
+                resolve
+            ).catch((error: Error) => {
+                this._logger.error(error);
+            });
+            this._removeOptions();
+        });
+    }
+
+    private closePopup(popup: string) {
+        Service.closePopup(popup);
+    }
+
+    public _ng_connectDialog() {
+        Service.requestPorts().then((response) => {
+            this._startSpy();
+            const popupGuid: string = this.api.addPopup({
+                caption: 'Choose port to connect:',
+                component: {
+                    factory: SidebarVerticalPortDialogComponent,
+                    inputs: {
+                        _onConnect: (() => {
+                            this._stopSpy().then(() => this._ng_onConnect());
+                            this.closePopup(popupGuid);
+                        }),
+                        _ng_canBeConnected: this._ng_canBeConnected,
+                        _ng_connected: this._ng_connected,
+                        _ng_onOptions: this._ng_onOptions,
+                        _ng_onPortSelect: this._ng_onPortSelect,
+                        _getSpyState: (() => this._ng_spyLoad),
+                        _requestPortList: ( () => response.ports),
+                        _getSelected: ((selected: IPortInfo) => { this._ng_selected = selected; }),
+                        _getOptionsCom: ((options: IOptions) => { this._optionsCom = options; }),
```

Comment:
sorry, why it's wrapped `()`?

Reply:
I thought it would need `()` that the whole function is assigned